open-build-service-api/src/lib.rs

```diff
@@ -408,6 +484,100 @@ impl<'a> PackageBuilder<'a> {
         self.client.request(u).await
     }
 
+    pub async fn source_file(&self, file: &str) -> Result<Vec<u8>> {
+        let mut u = self.client.base.clone();
+        u.path_segments_mut()
+            .map_err(|_| Error::InvalidUrl)?
+            .push("source")
+            .push(&self.project)
+            .push(&self.package)
+            .push(file);
+        Ok(
+            Client::send_with_error(self.client.authenticated_request(Method::GET, u))
+                .await?
+                .bytes()
+                .await?
+                .to_vec(),
+        )
+    }
+
+    pub async fn upload_for_commit<T: Into<Body>>(&self, file: &str, data: T) -> Result<()> {
+        let mut u = self.client.base.clone();
+        u.path_segments_mut()
+            .map_err(|_| Error::InvalidUrl)?
+            .push("source")
+            .push(&self.project)
+            .push(&self.package)
+            .push(file);
+        self.upload_file(file, Some("repository"), data).await?;
+        Ok(())
+    }
+
+    pub async fn commit(&self, filelist: &CommitFileList) -> Result<CommitResult> {
```

Comment:
this should potentially be a method on the Commit object instead; But really bigger open question here if it makes sense to directly expose this API.  Which in part is also an open question for this crate; What i did for the phabricator crate is to have one api crate which does really low-level api mangling (like this)..

For an initial implementation at this stage it's reasonable though;

Reply:
Also can you add some documentation on each function :) especially those where it's not clear to what OBS api it binds

Reply:
Is there a Commit struct I'm missing, or is that referring to some future design ideas?

> Also can you add some documentation on each function

Is this applying to the new functions, or all of the existing ones as well :sweat_smile:

Reply:
yes :p